### PR TITLE
Display HTTP request contents in the log in a long format

### LINKED ISSUE #8656: Display HTTP request contents in the log in a long format
It would be nice if we could enable a log of the request that comes to Quarkus and check the headers, method, path of the request, something like a verbose mode of curl.

## UPDATE
Access log is already implemented in #5556 / 7b9c133 and was released in 1.4.0.Final, so this issue is about displaying the HTTP contents in a long format, like:

```
GET /hello HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Connection: keep-alive
Host: localhost:8080
```
As suggested by @stuartwdouglas, the idea is to add a 'dump' or 'full' option to the config and an attribute that prints all the headers and have the 'dump' format string print the request line followed by the headers

### PR DESCRIPTION
**Pull request by hbelmiro**
[Issue 8656](https://github.com/quarkusio/quarkus/issues/8656#)

### CODE REVIEW

**Comment by hbelmiro** [comment]
I forgot to add the new pattern to the Javadoc in io.quarkus.vertx.http.runtime.AccessLogConfig#pattern. I'll commit it with the adjustments that may be requested in the review.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/filters/accesslog/AccessLogHandler.java [hunk 1]
```diff
@@ -104,12 +105,25 @@ public AccessLogHandler(final AccessLogReceiver accessLogReceiver, String format
...
+                        .add("%l")
+                        .add("%u")
+                        .add("%t")
+                        .add("\"%r\"")
+                        .add("%s")
+                        .add("%b")
+                        .add("%{ALL_REQUEST_HEADERS}")
+                        .toString();
+            default:
+                return formatString;
```
Comment (gastaldi) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/filters/accesslog/AccessLogHandler.java:125]: Excellent! :clap:

**Comment by gastaldi** [comment]
The output is a bit weird:

```
2020-07-13 17:39:23,414 INFO  [io.qua.htt.access-log] (executor-thread-1) 0:0:0:0:0:0:0:1
-
-
13/Jul/2020:17:39:23 -0300
"GET /greeting HTTP/1.1"
200
5
Host: localhost:8080
User-Agent: HTTPie/1.0.2
Accept-Encoding: gzip, deflate
Accept: */*
Connection: keep-alive
```


Can we get rid of the date and extra lines in the beginning? Also the quoted header?

**Comment by gastaldi** [comment]
It would be nice if it looked like this: 

```
GET /greeting HTTP/1.1
Host: localhost:8080
User-Agent: HTTPie/1.0.2
Accept-Encoding: gzip, deflate
Accept: */*
Connection: keep-alive

```

Also would it be possible to display the request body?

**Comment by gastaldi** [comment]
I'll give it a try ASAP. Can you squash your commits? Thanks!

**Review by gastaldi — APPROVED** [review]
I tested locally and it works like a charm! 

This is my configuration file: 
```properties
# Configuration file
quarkus.http.access-log.enabled=true
quarkus.http.access-log.pattern=long
```

And this is what appears in the logs: 

```
2020-07-21 11:08:03,810 INFO  [io.qua.htt.access-log] (executor-thread-1) 
GET /greeting HTTP/1.1
Host: localhost:8080
User-Agent: HTTPie/1.0.2
Accept-Encoding: gzip, deflate
Accept: */*
Connection: keep-alive
```

Good job @hbelmiro !

**Comment by hbelmiro** [comment]
Thank you for your help @gastaldi and @stuartwdouglas. It was great to make my first contribution to this project.

**Pull request merged**